js/common/lib/inference-session.ts

```diff
@@ -206,6 +209,19 @@ export declare namespace InferenceSession {
     deviceType?: 'cpu'|'gpu';
     powerPreference?: 'default'|'low-power'|'high-performance';
   }
+  export interface CoreMLExecutionProviderOption extends ExecutionProviderOption {
+    readonly name: 'coreml';
+    useCPUOnly?: boolean;
+    enableOnSubgraph?: boolean;
+    onlyEnableDeviceWithANE?: boolean;
+  }
+  export interface NnapiExecutionProviderOption extends ExecutionProviderOption {
+    readonly name: 'nnapi';
+    useFP16?: boolean;
+    useNCHW?: boolean;
+    cpuDisabled?: boolean;
+    cpuOnly?: boolean;
```

Comment:
I am not familiar with CoreML and NNAPI. but I want to know how is the options named? are they mapping to the corresponding config entries in each EP's definition?

If we have a more inconsistent naming it may help users to make less mistake. for example, in coreml we use `useCPUOnly` but in NNAPI we use `cpuOnly`. also, in NNAPI flags, it looks like that it's illegal for users to set `cpuDisabled` and `cpuOnly` both to `true`. maybe a string enum like:`mode?: 'cpu-only'|'defalut'|'gpu-only'` is better (again I am not familiar with NNAPI so this may not be appropriate, just an example)

Reply:
Also, are those configurations per-session or global?

If they are global, they should be put in `ort.env` instead of session options.

Reply:
It basically 1:1 mapping to native flags, I refer to [`coreml_provider_factory.h`](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/providers/coreml/coreml_provider_factory.h) and [`nnapi_provider_factory.h`](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/providers/nnapi/nnapi_provider_factory.h), and think it's easy let new user to understanding from the documentation. ([Core ML](https://onnxruntime.ai/docs/execution-providers/CoreML-ExecutionProvider.html#available-options) / [NNAPI](https://onnxruntime.ai/docs/execution-providers/NNAPI-ExecutionProvider.html#available-options))

> Also, are those configurations per-session or global?

It apply to session options, so it isn't global.

Reply:
They need to be in session options. Whether you want to use a specific EP can be model specific.